stellargraph/core/element_data.py

```diff
@@ -303,6 +305,37 @@ def _numpyise(d, dtype):
     return {k: np.array(v, dtype=dtype) for k, v in d.items()}
 
 
+class FlatAdjacencyList:
+    def __init__(self, flat_array, neighbor_counts):
+        self.splits = np.cumsum(np.append(0, neighbor_counts))
+        self.max_node_iloc = len(neighbor_counts) - 1
+        self.flat = flat_array
+        self.empty = np.array([], dtype=np.uint8)
+
+    def __getitem__(self, idx):
+        start, stop = self.splits[idx], self.splits[idx + 1]
+        return self.flat[start:stop]
+
+    def get(self, idx, default):
+        if idx > self.max_node_iloc:
+            return default
+        else:
+            return self[idx]
+
+    def items(self):
+        for idx in range(len(self.splits) - 1):
+            yield (idx, self[idx])
+
+
+def rle(sorted_arr):
```

Comment:
This name is a bit opaque, maybe either:

```suggestion
def run_length_encode(sorted_arr):
```

Or add a brief doc string (e.g. something like the one from my initial comment would be fine.

Also, this could potentially move to `utils.py`, I guess 🤷‍♂️